### PR TITLE
basic pretty printing, checking for "main"

### PR DESCRIPTION
**Pull request by Contentmaudlin**

### CODE REVIEW

File: src/backend/semant.ml [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
-       ast 
+
+open Ast
+open Astprint
+
+module GlobalMap = Map.Make(String)
+
+let check (ns, globals, functions) = 
+				(* Checking functions *)
+				if not (List.exists (fun fd -> (fd.fname = "main" && (Astprint.string_of_typ fd.ret_typ) = "int")) functions)
```
Comment (j-hui) [inline src/backend/semant.ml:10]: Keep in mind that later we will need to also check that the argument is of type `string[]`

File: src/backend/semant.ml [hunk 2]
```diff
@@ -1,3 +1,12 @@
 (* let's do some very basic checking here *) 
-let check ast = 
-       ast 
+
+open Ast
+open Astprint
```
Comment (j-hui) [inline src/backend/semant.ml:4]: where's the printer?

**Review by j-hui — APPROVED** [review]
the only thing is, where is `astpring.mly`?

File: src/backend/semant.ml (hunk 1 above)
Comment (Contentmaudlin) [reply src/backend/semant.ml:10]: Right. This should be contained in a general functions formals type checking .

File: src/backend/semant.ml (hunk 2 above)
Comment (Contentmaudlin) [reply src/backend/semant.ml:4]: frontend

**Pull request merged**